### PR TITLE
Switch from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by DreierF**
Addresses issue TS-25263

- [ ] Changes are tested adequately
- [ ] Teamscale documentation updated in case of relevant user-visible changes
- [ ] CHANGELOG.md updated
- [ ] Present new features in [N&N](https://wiki.cqse.eu/pages/viewpage.action?pageId=689566)

Please respect the vote of the [Teamscale bot](https://demo.teamscale.com) or flag irrelevant findings as tolerated or false positives. If you feel that the Teamscale config or architecture file needs adjustment, please state so in a comment and discuss this with your reviewer.

### CODE REVIEW

File: .github/workflows/actions.yml [hunk 1]
```diff
@@ -0,0 +1,82 @@
+name: Build
+
+on:
+  push:
+    branches: [ master ]
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:5]: i'm wondering whether we want to have builds before a PR is started on the branch in same repo?

File: .github/workflows/actions.yml [hunk 2]
```diff
@@ -0,0 +1,82 @@
...
+        name: target-linux
+        path: ./target
+        retention-days: 5
+
+  build-windows:
+    name: Windows Build
+    runs-on: windows-latest
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up GraalVM
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:39]: reuse with YAML anchor from build-linux? (e.g. `- *graal-setup`)

File: .github/workflows/actions.yml [hunk 3]
```diff
@@ -0,0 +1,82 @@
...
+        java-version: 11
+        graalvm-version: 20.3.0
+        native-image: true
+    - name: Install VS tools
+      run: choco install visualstudio2017-workload-vctools
+    - name: Build
+      env:
+        ACCESS_KEY: ${{ secrets.ACCESS_KEY }}
+      run: ./build-windows.bat
+    - name: 'Upload Artifact'
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:51]: reuse with YAML anchor from build-linux? (e.g. `- *upload-artifact`), might require setting a ENV var for `name` or maybe retrieve it from an already defined ENV var, e.g. CI_JOB (dunno if it exists)?

File: .github/workflows/actions.yml [hunk 4]
```diff
@@ -0,0 +1,82 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up GraalVM
+      uses: ayltai/setup-graalvm@v1
+      with:
+        java-version: 11
+        graalvm-version: 20.3.0
+        native-image: true
+    - name: Install VS tools
+      run: choco install visualstudio2017-workload-vctools
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:46]: do we need VS 2017? on windows-latest VS 2019 should be installed, if we need VS2017, switch to windows 2016? It's already installed there: https://github.com/actions/virtual-environments/blob/main/images/win/Windows2016-Readme.md

File: .github/workflows/actions.yml [hunk 5]
```diff
@@ -0,0 +1,82 @@
...
+  release:
+    if: startsWith(github.ref, 'refs/tags/v')
+    needs: 
+      - build-windows
+      - build-linux
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:68]: Do we need to checkout the code again?

File: pom.xml [hunk 6]
```diff
@@ -16,7 +16,7 @@
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
```
Comment (mpdeimos) [inline pom.xml:16]: shiver, MVN :vomiting_face:

**Review by mpdeimos — CHANGES_REQUESTED** [review]
Nice, GH actions look way more clean!

File: .github/workflows/actions.yml (hunk 2 above)
Comment (DreierF) [reply .github/workflows/actions.yml:39]: Unfortunately anchors are not yet supported https://github.com/actions/starter-workflows/issues/162

File: .github/workflows/actions.yml (hunk 3 above)
Comment (DreierF) [reply .github/workflows/actions.yml:51]: See comment above

File: .github/workflows/actions.yml (hunk 5 above)
Comment (DreierF) [reply .github/workflows/actions.yml:68]: Good point! It was need before I switch to the current release artifact upload action, but can be removed now

File: .github/workflows/actions.yml (hunk 1 above)
Comment (DreierF) [reply .github/workflows/actions.yml:5]: Sure, I can change it.

File: .github/workflows/actions.yml [hunk 7]
```diff
@@ -0,0 +1,78 @@
...
+    - uses: actions/checkout@v2
+    - name: Set up GraalVM
+      uses: ayltai/setup-graalvm@v1
+      with:
+        java-version: 11
+        graalvm-version: 20.3.0
+        native-image: true
+    - name: Build
+      env:
+        ACCESS_KEY: ${{ secrets.ACCESS_KEY }}
```
Comment (mpdeimos) [inline .github/workflows/actions.yml:24]: I've named the secret `TEAMSCALE_ACCESS_KEY` so it'S clear if you look at the admin page :)

**Pull request merged**